### PR TITLE
Apache Solr Update

### PR DESCRIPTION
**Pull request by daneshk**
This is to bundle apache solr version 4.7.2

### CODE REVIEW

File: solr/4.7.2.wso2v1/pom.xml [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+		WITHOUT WARRANTIES OR CONDITIONS OF ANY ~ KIND, either express or
+		implied. See the License for the ~ specific language governing
+		permissions and limitations ~ under the License.
+	-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+
+	<modelVersion>4.0.0</modelVersion>
+	<packaging>bundle</packaging>
+	<groupId>solr.wso2</groupId>
```
Comment (kasunbg) [inline solr/4.7.2.wso2v1/pom.xml]: The groupId should start with org.wso2.

File: pom.xml [hunk 2]
```diff
@@ -57,6 +57,7 @@
         <module>disruptor/3.2.1.wso2v1</module>
         <module>cassandra-thrift/1.2.13.wso2v4</module>
         <module>bsf/3.0.0.wso2v4</module>
+        <module>solr/4.7.2.wso2v1</module>
```
Comment (kasunbg) [inline pom.xml]: Please remove the parent pom reference.

File: solr/4.7.2.wso2v1/pom.xml [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+	</distributionManagement>
+
+	<build>
+		<plugins>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-compiler-plugin</artifactId>
+				<version>2.3.2</version>
+				<configuration>
+					<source>1.5</source>
```
Comment (kasunbg) [inline solr/4.7.2.wso2v1/pom.xml]: Can this be run on JDK 8 as well?

File: solr/4.7.2.wso2v1/pom.xml [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+ 							lucene-highlighter;scope=compile|runtime,
+							lucene-memory;scope=compile|runtime,
+							lucene-misc;scope=compile|runtime,
+							lucene-expressions;scope=compile|runtime,
+							lucene-grouping;scope=compile|runtime,
+							lucene-join;scope=compile|runtime,
+							lucene-spatial;scope=compile|runtime,
+							lucene-suggest;scope=compile|runtime,
+							lucene-analyzers-phonetic;scope=compile|runtime,
+							commons-io;scope=compile|runtime,
```
Comment (kasunbg) [inline solr/4.7.2.wso2v1/pom.xml]: Why are you embedding a commons-io dependency here? We already have a commons-io bundle with version 2.0.0 (commons-io 2.0.0.wso2v2).

If you need a later version, please create a new separate orbit bundle for commons-io

File: solr/4.7.2.wso2v1/pom.xml [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+							httpmime;scope=compile|runtime,
+							zookeeper;scope=compile|runtime,
```
Comment (kasunbg) [inline solr/4.7.2.wso2v1/pom.xml]: For zookeeper - same as the commons-io scenario above. Thanks

File: solr/4.7.2.wso2v1/pom.xml [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+							noggit;scope=compile|runtime,
+							spatial4j;scope=compile|runtime
+						</Embed-Dependency>
+						<Export-Package>
+							org.apache.solr.*;version="${solr.version}",
+							org.tartarus.*
```
Comment (kasunbg) [inline solr/4.7.2.wso2v1/pom.xml]: In what ways does the org.tartarus package is relevant to solr for it to be include as part of solr orbit?

File: solr/4.7.2.wso2v1/pom.xml [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+							jline; resolution:=optional,
+							org.apache.commons.codec.*;resolution:=optional,
+							org.apache.hadoop.*; resolution:=optional,
+							org.apache.lucene.sandbox.queries; resolution:=optional,
+							org.eclipse.jetty.*; resolution:=optional,
+							org.jboss.netty.*; resolution:=optional,
+							org.objectweb.asm.*; resolution:=optional,
+							org.restlet.*; resolution:=optional,
+						</Import-Package>
+						<DynamicImport-Package>*</DynamicImport-Package>
```
Comment (kasunbg) [inline solr/4.7.2.wso2v1/pom.xml]: Does Solr need Dynamic-Import?

File: httpclient/4.3.1.wso2v1/pom.xml [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+                        <Bundle-SymbolicName>${pom.artifactId}</Bundle-SymbolicName>
+                        <Bundle-Name>${pom.artifactId}</Bundle-Name>
+                        <Export-Package>
+                            org.apache.http.*;version="${version.httpclient}"
+                        </Export-Package>
+                        <Import-Package>
+                            !org.apache.http.*,
+                            org.ietf.jgss; version="0.0.0",
+                            org.apache.commons.logging; version="1.2.0",
+                            org.apache.commons.codec.binary; version="1.4.0",
```
Comment (Kishanthan) [inline httpclient/4.3.1.wso2v1/pom.xml]: Import versions should have version range. Eg : [1.4.0, 1.5.0).

File: lucene/5.0.0.wso2v1/pom.xml [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+                        <Embed-Dependency>
+                            lucene-codecs;scope=compile|runtime,
+                            lucene-analyzers-phonetic;scope=compile|runtime,
+                            lucene-suggest;scope=compile|runtime,
+                            lucene-backward-codecs;scope=compile|runtime,
+                            lucene-analyzers-kuromoji;scope=compile|runtime
+                        </Embed-Dependency>
+                        <Export-Package>
+                            org.apache.lucene.*;version="${version.lucene}"; -split-package:=merge-first,
+                            org.tartarus.snowball.*
```
Comment (Kishanthan) [inline lucene/5.0.0.wso2v1/pom.xml]: Need a concrete version on exporting packages.

File: noggit/0.6.wso2v1/pom.xml [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+    <description>
+        This bundle will export packages from noggit jar, org.noggit.*
+    </description>
+    <url>http://wso2.org</url>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.noggit</groupId>
+            <artifactId>noggit</artifactId>
+            <version>${version.noggit}</version>
```
Comment (Kishanthan) [inline noggit/0.6.wso2v1/pom.xml]: Make this an optional dependency.

Comment (daneshk) [inline noggit/0.6.wso2v1/pom.xml]: Marked all dependencies as optional

**Pull request merged**